modules/mini-games/tightspot.lua

```diff
@@ -393,52 +433,60 @@ local function placed_entety(event)
         end
     end
 end
-local token_for_replace_wall
+
+--- Used to replace wall after they have been mined
+local replace_wall = Token.register(function(args)
+    variables["surface"].create_entity(args[1])
+    args[2].remove_item{name = args[1].name, count = 1}
+end)
+
+--- Triggered when an entity is mined
 local function mined(event)
     local entity = event.entity
-    if entity.type == "wall" then
-        local position = entity.position
-        local player = game.players[event.player_index]
-        if not variables.walls[player.name][position.x .. "," .. position.y] then
-            local force = entity.force
-            task.set_timeout_in_ticks(
-                1,
-                token_for_replace_wall,
-                {{name = entity.name, position = position, force = force}, player.name}
-            )
-            player.print("How did you get here. (Ps if not command uses tell and admin of this.)")
-        else
-            local price = land_price(player, position)
-            player.surface.create_entity {
-                name = "flying-text",
-                position = position,
-                text = "-" .. price,
-                color = {r = 1}
-            }
-            change_balance(player, price * -1)
-        end
+    if entity.type ~= "wall" then return end
+    local position = entity.position
+    local player = game.players[event.player_index]
+    if not variables.walls[player.name][position.x .. "," .. position.y] then
+        -- If the player was not the one to place it then replace it
+        task.set_timeout_in_ticks(
+            1, replace_wall,
+            {{name = entity.name, position = position, force = entity.force}, player}
+        )
+        player.print("How did you get here. (Ps if not command uses tell and admin of this.)")
+    else
+        -- Otherwise give the player money
```

Comment:
this not giving money its taking money 😛

Reply:
Will correct the comment